### ISSUE TITLE
TCMR : Beyond Static Features for Temporally Consistent 3D Human Pose and Shape from a Video

### ISSUE THREAD

**Issue by Songinpyo** [Songinpyo/Paper-Record#20]
## ❓ Why → 해당 논문의 필요성, abstract
기존의 video-based methods는 현재 프레임의 static feature에 storng dependency를 가지는 image based method's temporal inconsistency를 해결하는데 실패하였다.
우리가 제시하는 TCMR은 효과적으로 과거와 미래의 frame의 temporal information에 집중하며 현재 프레임의 static feature에 dominated되지 않는다.

🌟 **Contribution**
1. TCMR은 과거와 현재 frame의 temporal information을 효과적으로 활용하며 현재 프레임의 static feature에 지배당하지 않는다.
2. TCMR은 Temporal consistency를 강화할 뿐 아니라 per-frame 3D pose shape inference도 개선하였다.

## 💡 Key insights → 해당 논문의 핵심 아이디어
![image](https://user-images.githubusercontent.com/104220612/189799451-18059361-7a7c-4ce8-943d-55896d905062.png)

Current frame 즉, target frame은 input frame T의 T/2 frame으로 정해진다.
2개의 Uni-directional GRU를 통해서 static features로부터 반대 방향으로 temporal features를 추출한다.
각각 f1, fT 양 끝에서부터 fT/2까지 Uni-directional GRU를 통해서 g_all feature를 구한다.
이후에 1~T/2-1, T/2+1~T 까지 각각 GRU를 통해 g_past, g_future를 구한다.
이렇게 만들어진 3개의 g는 차원 조정후에 self-attention을 통해서 중요도를 구하게 되고, 중요도를 곱한 값을 element-wise sum 하여 최종 g_int를 구한다.

📌 **Temporal encoding from all frames**
g_all 을 구하는 과정으로, 위에서 서술한대로 Uni-directional GRUs를 통해서 구한다.

📌 **Temporal encoding by PoseForecast**
Past :  1~(T/2)-1
Future :  (T/2)+1~T
로 범위를 잡아서 Current frame을 제외한 temporal feature g_past와 g_future를 구한다.

📌 **Temporal feature integration**
![image](https://user-images.githubusercontent.com/104220612/189824776-2654d9cb-07ba-4e8e-b699-32ffbb817fe9.png)
Temporal feature를 ReLU 활성화함수를 통과시키고 Fully connected layer를 통해 2048->256 으로 resize한다.
3개의 256 feature (g`_past, g`_all, g`_future)를 concat하고 self-attention을 통해서 중요도를 구한다.
![image](https://user-images.githubusercontent.com/104220612/189825488-fa96ab18-884a-4dcb-97fe-3c9c50829dba.png)
이후 중요도와 feature를 곱한후 elemetal-wise sum을 통해서 최종 output g_int를 구한다.

Training 시에는 g`_past, g_int, g`_future가 모두 하나의 SMPL parameter regressor를 통해서 Θ_past, Θ_int, Θ_future를 구하게 된다.
각 Θ는 θ, β, s, t 를 포함한다. (s, t 는 camera parameter)

즉, 미래와 과거의 Θ를 통해서도 loss를 구하여 학습하는 듯 하다.

Test 시에는 g_int만 parameter를 통과하여 Θ_int 가 최종 3D human mesh output이 된다.

📌 **Loss functions**
Training시에는 Θ_past, Θ_int, Θ_future 모두 current frame groundtruth 를 통해 학습된다.
SMPL parameters, 2D/3D joint coordinates들의 L2 loss를 구한다.

3D joint coordinates는 예측한 camera parameters과 SMPL parameter를 통해서 구하고 2D joint coordinates는 3D를 projection하여 구한다.

📌 **Performance**
![image](https://user-images.githubusercontent.com/104220612/189826785-a20e0ba3-ab68-427b-b03f-a3dbd29a993e.png)
현재 프레임의 의존성을 줄임으로써 accel이 줄어든 것은 당연하지만, per frame performance가 개선된점은 주목할만하다.

## 🔗 Reference URL → 논문 링크
[TCMR : Beyond Static Features for Temporally Consistent 3D Human Pose and Shape from a Video](https://arxiv.org/abs/2011.08627)
